docs/proposal/1-authentication-support/README.md

```diff
@@ -0,0 +1,175 @@
+kind: Secret
+metadata:
+  name: kstone-api-admin
+  namespace: kstone
+type: Opaque
+```
+
+To customize admin username and password, modify the user and password field in `kstone-api-admin` secret.
+
+#### Encryption && Decryption
+
+Use [`golang.org/x/crypto`](https://pkg.go.dev/golang.org/x/crypto) to compare a **bcrypt hashed password** with its **possible plaintext equivalent**. Returns nil on success, or an error on failure.
+
+- **bcrypt hashed password** stored in `kstone-api-admin` secret.
+- **possible plaintext equivalent** is the login password from user.
+```go
+bcrypt.CompareHashAndPassword(bcryptHashedPassword, plaintextPassword)
+```
+
+#### Customize Admin Username, Password and Private Key
+
+- Specify a new username and password(e.g. admin:admin)
+```shell
+USER=admin
+PASSWORD=admin
+```
+
+##### Customize Password
+- get password hash and encode into base64:
+
+```shell
+# yum -y install httpd-tools
+# e.g. htpasswd -nbBC 10 admin admin
+HASH_PASSWORD=`htpasswd -nbBC 10 ${USER} ${PASSWORD}|awk -F ':' '{print $2}'`
+
+```
+
+##### Customize Private Key
+
+```shell
+openssl genrsa -out private.key 2048
+```
+
+##### Print out Custom kstone-api-admin Secret
+- use `--dry-run=client` option to print out a custom `kstone-api-admin` secret:
+```shell
+kubectl create secret generic -n kstone kstone-api-admin --from-file=private.pem --from-literal=user=${USER} --from-literal=password=${HASH_PASSWORD}  --dry-run=client -oyaml
+```
+
+- replace the default `kstone-api-admin` secret with the new `kstone-api-admin` secret.
+
+
+### External Auth
+
+1. Implement the Request interface to verify the external authentication request.
```

Comment:
Here we can introduce what we plan to support authentication plugin in the future, such as what work should be done if we want to support webhook.